core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -228,21 +369,41 @@ export class ClassroomAdminPageComponent implements OnInit {
       name: classroomDict.name,
       url_fragment: classroomDict.urlFragment,
       course_details: classroomDict.courseDetails,
+      teaser_text: classroomDict.teaserText,
       topic_list_intro: classroomDict.topicListIntro,
       topic_id_to_prerequisite_topic_ids:
         classroomDict.topicIdToPrerequisiteTopicIds,
+      is_published: classroomDict.isPublished,
+      thumbnail_data: classroomDict.thumbnailData,
+      banner_data: classroomDict.bannerData,
     };
   }
 
+  unpublishClassroom(): void {
+    this.classroomDataUnpublishInProgress = true;
+    this.tempClassroomData.setIsPublished(false);
+    this.updateClassroomData(this.tempClassroomData.getClassroomId());
```

Comment:
Here and below, it looks like updateClassroomData is an async function. In that case, should the unsetting of flag be done only on completion of the async call?

Reply:
yes makes sense. Done.